Inferences/ForwardDemodulation.cpp

```diff
@@ -151,19 +150,16 @@ bool ForwardDemodulationImpl<combinatorySupSupport>::perform(Clause* cl, Clause*
         } else {
           rhsS = subs->applyToBoundResult(rhs);
         }
-        if(resultTermIsVar){
```

Comment:
This looks like an interesting change. Do you remember, why we don't need this line anymore?

Reply:
Tbh I'm not sure whether I did this intentionally. I reverted it now anyways.